zilliqa/src/api/zil.rs

```diff
@@ -103,10 +112,70 @@ fn create_transaction(params: Params, node: &Arc<Mutex<Node>>) -> Result<serde_j
         sig,
     };
 
-    let transaction_hash = node.create_transaction(transaction)?;
-    let transaction_hash = hex::encode(transaction_hash.0);
+    let transaction_hash = node.create_transaction(transaction.clone())?;
+    let transaction_verified: VerifiedTransaction = transaction.verify()?;
+
+    match transaction_verified.tx {
+        SignedTransaction::Zilliqa { ref tx, .. } => {
+            let contract_address = get_created_scilla_contract_addr(tx, transaction_verified.signer);
+            let response = CreateTransactionResponse{contract_address: contract_address, info: "Txn processed".to_string(), tran_id: transaction_hash.0.into()};
+            trace!("CreateTransaction: {:?} response: {}", tx, serde_json::to_string(&response).unwrap());
+            Ok(response)
+        }
+        _ => {
+            Err(anyhow!("unexpected transaction type for scilla create transaction"))
+        }
+    }
+}
+
+fn get_transaction(params: Params, node: &Arc<Mutex<Node>>) -> Result<Option<GetTxResponse>> {
+    let hash: H256 = params.one()?;
+    let hash: Hash = Hash(hash.0);
+
+    let ret = get_scilla_transaction_inner(hash, &node.lock().unwrap())?;
+    let mut receipt = node.lock().unwrap().get_transaction_receipt(hash)?;
+
+    // Spin up to 1s while waiting for the receipt to become available
```

Comment:
There is some issue with the JS framework/scilla plugin where it attempts to immediately query the TX receipt after submitting the TX. Responding null does not work. This causes the test to erronerously fail - I am looking into it.

Reply:
I'd feel quite uncomfortable adding this code to `main` since it seems like a serious performance problem waiting to happen.

We should check the behaviour of the ZQ1 API when this method is called for a transaction that hasn't committed yet.